### ISSUE TITLE
Allow modification of existing permissions in ChannelAction

### ISSUE THREAD

**Issue by ScoreUnder** [DV8FromTheWorld/JDA#1499]
Inspired by discord/discord-api-docs#2520

## General Troubleshooting
  
- [X] I have checked for similar issues.
- [X] I have updated to the [latest JDA version][download].
- [X] I have checked the branches or the maintainers' PRs for upcoming features.

## Feature Request

I would like a way to view and update existing permission overrides from a `ChannelAction` object.

### Example Use-Case

Because of discord/discord-api-docs#2520, when using JDA's `Category#createVoiceChannel` you must now sanitise out all `MANAGE_ROLES` permissions unless the bot is given admin permissions. Currently I'm just using this as a quick hack, which is not ideal:

```scala
private object JDAHack {
  lazy val field: Field = {
    val f = classOf[ChannelActionImpl[_]].getDeclaredField("overrides")
    f.setAccessible(true)
    f
  }

  def removeManageRolesPerms(channelReq: ChannelAction[_]): Unit = {
    try {
      val overrides = field.get(channelReq).asInstanceOf[TLongObjectHashMap[PermOverrideData]]
      overrides.transformValues { v =>
        val manageRoles = Permission.MANAGE_ROLES.getRawValue
        new PermOverrideData(v.`type`, v.id, v.allow & ~manageRoles, v.deny & ~manageRoles)
      }
    } catch {
      case NonFatal(_) =>
    }
  }
}
```

**Comment by Andre601** [comment]
I'm not an expert, but when you create the channel should you be able to set allow and deny using setAllow/setDeny.

Alternatively can you do `VoiceChannel#getManager().upsertPermission(...)` in the success callback of `queue()` from the create action.

Tl;dr: there are already ways to change a channel's permission individually and your "hack" is redundant.

**Comment by ScoreUnder** [comment]
> but when you create the channel should you be able to set allow and deny using setAllow/setDeny.

But I still have no control over permissions applied automatically by `Category#createVoiceChannel`

> in the success callback of `queue()` from the create action.

It can never succeed on a non-admin user when this permission is in the action

**Comment by MinnDevelopment** [comment]
You can use [`Guild#createVoiceChannel(name, parent)`](https://github.com/DV8FromTheWorld/JDA/blob/ac6fc8d4c93593cc134825d6cfb7b4ec7453b1b1/src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java#L1343) which starts off with no overrides at all.

**Comment by MinnDevelopment** [comment]
You can also use `ChannelAction#addPermissionOverride` to replace existing overrides.

**Comment by ScoreUnder** [comment]
I considered those, but using `createVoiceChannel` on the guild leaves me the problem of not taking permissions from the category (e.g. voice ban and priority speaker roles), and replacing with `addPermissionOverride` requires that I know exactly which permissions have that kill-bit in them in the first place.
